Makefile

```diff
@@ -0,0 +1,227 @@
+# perform validation to avoid many combination provided simultaneously
+override BUMP_VERSION_INPUT := 1
+override BUMP_VERSION_LEVEL :=
+ifeq ($(filter major, $(MAKECMDGOALS)), major)
+  ifeq ($(BUMP_VERSION_LEVEL),)
+    override BUMP_VERSION_LEVEL := major
+    override BUMP_VERSION_INPUT := 0
+  else
+    $(error "Cannot combine 'major|minor|patch' simultaneously!")
+  endif
+endif
+ifeq ($(filter minor, $(MAKECMDGOALS)), minor)
+  ifeq ($(BUMP_VERSION_LEVEL),)
+    override BUMP_VERSION_LEVEL := minor
+    override BUMP_VERSION_INPUT := 0
+  else
+    $(error "Cannot combine 'major|minor|patch' simultaneously!")
+  endif
+endif
+ifeq ($(filter patch, $(MAKECMDGOALS)), patch)
+  ifeq ($(BUMP_VERSION_LEVEL),)
+    override BUMP_VERSION_LEVEL := patch
+    override BUMP_VERSION_INPUT := 0
+  else
+    $(error "Cannot combine 'major|minor|patch' simultaneously!")
+  endif
+endif
+ifneq ($(VERSION),)
+  ifneq ($(BUMP_VERSION_LEVEL),)
+    $(error "Cannot combine 'major|minor|patch' simultaneously with explicit VERSION input!")
+  endif
+endif
+# when none was provided, use patch to directly apply the explicit 'VERSION' specified as input
+ifeq ($(BUMP_VERSION_LEVEL),)
+  override BUMP_VERSION_LEVEL := --new-version "$(VERSION)" patch
+endif
+
+.PHONY: dry
+dry: $(BUMP_CFG)	## Run 'bump' without applying changes (dry-run) (call: make VERSION=<MAJOR.MINOR.PATCH> bump dry)
+	@-$(ECHO) > /dev/null
+
+.PHONY: patch
+patch: $(BUMP_CFG)	## Run 'bump' with automatic semantic versioning of the next PATCH number (call: make bump patch)
+	@-$(ECHO) > /dev/null
+
+.PHONY: minor
+minor: $(BUMP_CFG)	## Run 'bump' with automatic semantic versioning of the next MINOR number (call: make bump minor)
+	@-$(ECHO) > /dev/null
+
+.PHONY: major
+major: $(BUMP_CFG)	## Run 'bump' with automatic semantic versioning of the next MAJOR number (call: make bump major)
+	@-$(ECHO) > /dev/null
+
+# run bumpversion with replacement of the release time value within its config for auto-update on future revisions
+.PHONY: bump
+bump: bump-check bump-install  ## Bump version using specified <VERSION> (call: make VERSION=<MAJOR.MINOR.PATCH> bump)
+	@-$(MSG_I) "Updating package version..."
+	@[ $(BUMP_VERSION_INPUT) -eq 0 ] || [ "${VERSION}" ] || ( \
+		$(MSG_E) "Argument 'VERSION' is not specified to bump version"; exit 1 \
+	)
+	@$(SHELL) -c ' \
+		PRE_RELEASE_TIME=$$(head -n 1 RELEASE.txt | cut -d " " -f 2); \
+		$(CONDA_CMD) $(BUMP_CMD) $(BUMP_XARGS) $(BUMP_VERSION_LEVEL); \
+		POST_RELEASE_TIME=$$(head -n 1 RELEASE.txt | cut -d " " -f 2); \
+		echo "Replace $${PRE_RELEASE_TIME} → $${POST_RELEASE_TIME}"; \
+		$(_SED) -i "s/$${PRE_RELEASE_TIME}/$${POST_RELEASE_TIME}/g" $(BUMP_CFG); \
+		git add $(BUMP_CFG); \
+		git commit --amend --no-edit \
+	'
+
+.PHONY: bump-install
+bump-install:   ## Installs bumpversion if not detected in the environment
+	@-$(SHELL) -c '$(CONDA_CMD) test -f "$(BUMP_PATH)" || pip install $(PIP_XARGS) $(BUMP_TOOL)'
```

Comment:
Using `conda` to test but `pip` to install? :D  Just curious, why just not stay with `conda`?

Another option is to create an `environment.yml` file with this kind of content:
```
# conda env create -f environment.yml
name: birdhouse-deploy
channels:
  - conda-forge
  - defaults

dependencies:
  - make
  - bump2version
```

And just tell user to run it and activate it before running the various `make bump` targets.

Reply:
The `CONDA_CMD` allows you to define for example `CONDA_CMD = source ${CONDA_HOME}/bin/activate ${CONDA_ENV};` which would automatically activate that specific env before running any operation to make sure it installs in the right location. If left blank, it will simply run the command in whichever env that is already active (or system/conda base env if none was). So it is not using conda to test (not `conda test`), just the shell `test` command for locating the binary of `bumpversion`.